### PR TITLE
Request Mirroring

### PR DESCRIPTION
**Pull request by matined**

### CODE REVIEW

File: app/api/services/model_mirroring.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# main.py
```
Comment (HermanPlay) [inline app/api/services/model_mirroring.py:1]: This is not a main.py file

File: app/api/charts/pool/templates/mirror-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      app: tyro-model-mirror-{{ .Values.name }}
+  template:
+    metadata:
+      labels:
+        app: tyro-model-mirror-{{ .Values.name }}
+    spec:
+      containers:
+        - name: tyro-model-mirror-{{ .Values.name }}
+          #  to do: replace with your our docker image containing mirroring handler
+          image: your_docker_username/your_fastapi_app:v1
```
Comment (HermanPlay) [inline app/api/charts/pool/templates/mirror-deployment.yaml:19]: This should be the image of the model, right?

**Review by HermanPlay — CHANGES_REQUESTED** [review]
Interesting idea

**Review by matined — COMMENTED** [review]
That a nice approach, but needs a discussion.

If we want a custom mirroring api, we can move the container outside of the api, as we need to build the image separately. 

If we have a separate mirroring api for each pool we can pass a list of shadow models as a environmental variable.

File: app/api/utils/deployments/pool.py [hunk 3]
```diff
@@ -55,12 +55,9 @@ def __make_values(self) -> dict:
                     }
                     for model in self.models.targets
                 ],
-                "shadows": [
-                    {
-                        "name": self.__make_host(model),
-                    }
-                    for model in self.models.shadows
-                ],
+                "shadows": ",".join(
```
Comment (matined) [inline app/api/utils/deployments/pool.py:58]: I dont like it :// Can't we just pass a list like in targets?

**Pull request closed without merging**